### ISSUE TITLE
Request KWh reading

### ISSUE THREAD

**Issue by daOoZF8** [jasonacox/powermonitor#1]
Thanks.  Work great.  Is it possible to add KWh?  It will be great to capture total power used over time.  Thanks.

**Comment by Banana5kin** [comment]
Yeah, that would be a great addition.

**Comment by jasonacox** [comment]
This is a basic script that pulls a snapshot of power measured through the plug.  Since there is no memory on the device itself, to get the time dimension, the script would need to be set up to sample the KW values at intervals and record the timestamp and values.  It could then use those data points compute the area under the curve to give the total KWh value since the first sample.  Alternatively, if you just want a crude estimate of KWh for a month based on a single sample (assuming the power usage stays constant) the script could simply multiply by the hours in the month.

**Comment by jasonacox** [comment]
I added kWh in the output, projected out for Day, Week and Month.  Keep in mind these are estimates based on the point in time snapshot (assumes power stays constant).  If someone wants to extend this for a cron script that samples at intervals and computes the actual power usage, please feel free to do so and submit a pull request.

**Comment by misroka** [comment]
Hi, I'm using one plug that normally collects information about consumption: https://nedis.com/en-us/product/security-safety/smart-home/power/550671326/smartlife-smart-plug-wi-fi-power-meter-2500-w-plug-with-earth-contact-type-f-cee-77-10-40-0c-android-ios-white 

Do you know if it is possible to read these values somehow? I believe there is some memory on the plug.

**Comment by jasonacox** [comment]
Yes, I believe there are now devices with some memory to compute energy over durations (kWh).  They would show up in the DPS (data points) that come back from the device.  The core library for this project is [TinyTuya](https://github.com/jasonacox/tinytuya#tinytuya) and I recommend you take a look at the DPS index your device has.  It may be providing the energy data you need. Here is how to find out:

```python
import tinytuya

d = tinytuya.OutletDevice(DEVICEID, DEVICEIP, DEVICEKEY)
d.set_version(3.3)

print(" > Fetch DPS Values < ")
data = d.status()
print(data)
```

Some helpful documentation on DPS is [here](https://github.com/jasonacox/tinytuya#tuya-data-points---dps-table).  You may need to use the Tuya Cloud and trial/error to determine the mapping of the DPS values to what they mean.  Please post what you learn.  It may help others on the research as well.
